Fix issues in release automation

ankita240796 opened the pull request:
##  Explanation
Fixes issues in release automation:

- Fixes message in restore backup script
- Fixes path in all release scripts
- Updates deploy script to allow test branch deployment only on backup server

## Checklist
- [ ] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [ ] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The linter/Karma presubmit checks have passed.
  - These should run automatically, but if not, you can manually trigger them locally using `python -m scripts.pre_commit_linter` and `python -m scripts.run_frontend_tests`.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR has an appropriate "PR CHANGELOG: ..." label (If you are unsure of which label to add, ask the reviewers for guidance).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide).
- [x] The PR addresses the points mentioned in the codeowner checks for the files/folders changed. (See the [codeowner's wiki page](https://github.com/oppia/oppia/wiki/Oppia%27s-code-owners-and-checks-to-be-carried-out-by-developers).)
- [x] The PR is **assigned** to an appropriate reviewer.
  - If you're a new contributor, please ask on [Gitter](https://gitter.im/oppia/oppia-chat) for someone to assign a reviewer and don't tick this checkbox.
  - If you're not sure who the appropriate reviewer is, please assign to the issue's "owner" -- see the "talk-to" label on the issue. Do not only request the review but also add the reviewer as an assignee.

ankita240796 committed:
Fix paths

ankita240796 committed:
Update deploy script to avoid test branch deployment on test server

ankita240796 committed:
Fix message

ankita240796 committed:
Fix name

seanlip reviewed (approved):
LGTM. Thanks @ankita240796!

DubeySandeep commented on scripts/release_scripts/deploy.py:
```diff
@@ -23,13 +23,13 @@
     running
 
-        python -m scripts.install_third_party_libs
+        python -m scripts.release_scripts.install_third_party_libs
```
I think it's in the `scripts` dir as this is used in running the server and setup.

ankita240796 replied:
My bad, fixed.

You have become a great reviewer btw, I really need to learn these skills 😛

DubeySandeep reviewed (approved):
@ankita240796, Thanks for fixing these minor issues, I've left one comment. Feel free to merge this PR once that comment is resolved. Thanks!

ankita240796 committed:
Merge remote-tracking branch 'upstream/develop' into fix-release

ankita240796 committed:
Revert wrong change

ankita240796 committed:
Merge remote-tracking branch 'upstream/develop' into fix-release

ankita240796 committed:
Fix test

ankita240796 committed:
Merge remote-tracking branch 'upstream/develop' into fix-release

seanlip commented on scripts/release_scripts/initial_release_prep_test.py:
```diff
@@ -135,56 +133,59 @@ def mock_run_cmd(cmd_tokens):
             python_utils, 'INPUT', mock_input)
-        with open_tab_swap, check_call_swap, input_swap:
+        with open_tab_swap, cwd_swap, input_swap, self.assertRaisesRegexp(
+            Exception, 'Exception raised to stop branch cut.'):
```
What's catching these exceptions?

Also, why do we need to raise them? Seems like an odd pattern.

ankita240796 replied:
`self.assertRaisesRegexp` catches the Exception here.

Some context on why I made this change ([Ref](https://github.com/oppia/oppia/pull/8577#discussion_r379821699)): 

- I updated the arg passed in `cut_release_or_hotfix_branch` from `new_version` to `release_version` in PR #8577 
- I missed changing the subprocess call in `initial_release_prep` and the tests did not catch it since I was mocking `subprocess.check_call` in tests.
- So, I need to actually run the `cut_release_or_hotfix_branch` script instead of mocking `subprocess.check_call` in the tests but also ensure that branch cut is not executed. 
- If I mocked the `execute_branch_cut` function which is being called, it will again fail to detect if a wrong arg is passed.
- So, I wanted the function to run but stop as soon as arg check is done. The first line after arg check in the function is to check if the cwd is oppia. So, I mocked that function instead and raised an Exception there to ensure that args are checked as well as the function stops execution after that.

\cc @DubeySandeep

ankita240796 replied:
You can ignore the previous comment. I have modified the branch cut script by adding a main method as per @DubeySandeep's suggestion and now there is no need for exception.

DubeySandeep commented on scripts/release_scripts/cut_release_or_hotfix_branch.py:
```diff
@@ -226,10 +227,10 @@ def _get_hotfix_branch_type_and_name(target_version, hotfix_number):
 
 
-def execute_branch_cut():
+def execute_branch_cut(args=None):
```
I was thinking of having explicit params here and the `main` method should parse the args and pass the params to this method. That way other scripts (which are using this script) will be independent of the keyword used for args. Let me know what you think?

Also, do we need to add `Args` docstring here? (I was expecting linters to catch this.)

ankita240796 replied:
Yes, I agree. Adding a main method will make the test easy as well. I have added a main method now.

One doubt:
> That way other scripts (which are using this script) will be independent of the keyword used for args.

We will still need to pass the arg even if main method parses it,  how will they be independent of the names used for args?

> Also, do we need to add Args docstring here? (I was expecting linters to catch this.)

Not sure, but I do not see docstring for any of the case with `args=None`.

DubeySandeep commented on scripts/release_scripts/deploy.py:
```diff
@@ -465,8 +465,8 @@ def execute_deployment():
     if common.is_current_branch_a_test_branch() and (
-            app_name == APP_NAME_OPPIASERVER):
-        raise Exception('Test branch cannot be deployed to prod.')
+            app_name in [APP_NAME_OPPIASERVER, APP_NAME_OPPIATESTSERVER]):
```
DO we have any APP_NAME_* for the backup server as well?

ankita240796 replied:
No, we don't.

DubeySandeep reviewed (commented):
@ankita240796, Thanks for fixing the `release_version` test issue, I've 2 questions other than that this PR LGTM!

ankita240796 committed:
Fix functions

ankita240796 committed:
Merge remote-tracking branch 'upstream/develop' into fix-release

ankita240796 committed:
Add comment

seanlip commented on scripts/release_scripts/cut_release_or_hotfix_branch_test.py:
```diff
@@ -325,23 +325,21 @@ def test_missing_release_version(self):
                 with self.verify_target_version_swap, self.open_tab_swap:
                     with self.get_remote_alias_swap, self.check_call_swap:
-                        with input_swap, args_swap, self.assertRaises(
+                        with input_swap, self.assertRaises(
```
Everywhere, use assertRaisesRegexp and include an error message, to verify that the exception is being raised in the correct context.

seanlip replied:
(I've also added this to #8423 as a lint check request.)

ankita240796 replied:
A `system.exit` is raised here, so there is not specific message to check.

seanlip replied:
Where is the SystemExit raised from? I think you should be able to add a message to it explaining what went wrong...

ankita240796 replied:
Updated.

seanlip reviewed (approved):
Thanks @ankita240796! LGTM, just had one note.

vojtechjelinek committed:
Reduce bundle size and tidy up manifest (#8663)

* Move angular cookies from manifest to package

* Load bootstrap through webpack

* Move bootstrap to webpack

* Other work

* Remove bootstrap using webpack

* Remove libs

* Fix frontend tests

* Fix frontend tests

* Fix yarn

* Add comment

* Use regular FA

nishantwrp committed:
Fix #8602: Fetch the messages after updating the suggestion (#8624)

* Fixed the labels in suggestion review modal

* Revert "Fixed the labels in suggestion review modal"

This reverts commit 8624c77b575d45e4829127254ab66adc526c523d.

* Fetch messages again when updated the thread

* Changes

* Return

* TODO

* Lint

* Fix e2e tests

U8NWXD committed:
Fixes part of #6240: Fix Flaky Exploration Card Finding (#8659)

* Protractor: Fix Flaky Exploration Card Finding

Finding exploration cards on the creator dashboard page was flaky
because sometimes the protractor test would search for the cards before
the cards had loaded. Further, the logic used to identify matching cards
failed when the card contained text besides the exploration title. This
commit resolves these issues by:

* Waiting for an exploration card to appear before searching for them
* When filtering cards by title, consider cards to match if their text
  equals the target title or begins with the target title concatenated
  with a newline. This works because when other text is included in the
  card, a newline separates that text from the title.

* Style fixes

* Replace presenceOf with visibilityOf

We were waiting for a card to exist in the DOM (whether or not it was
visible), but we really do want it to be visible before clicking it.
Further, by relying on `visibilityOf`, we can use existing code. This
commit makes this switch.

ankita240796 committed:
Merge branch 'fix-release' of github.com:ankita240796/oppia into fix-release

ankita240796 committed:
Add proper exception

ankita240796 commented:
Closing this one, commits got messed up accidentally.
